pkg/automationconfig/automation_config_builder.go

```diff
@@ -221,22 +236,36 @@ func (b *Builder) Build() (AutomationConfig, error) {
 		dataDir = b.dataDir
 	}
 
-	totalVotes := 0
 	for i, h := range hostnames {
+		// Arbiters start counting from b.members and up
+		isArbiter := i >= b.members
+		replicaSetIndex := i
+		processIndex := i
+
+		if isArbiter {
+			processIndex = i - b.members
+			// The arbiter's index will start on `arbitersStartingIndex` and increase
+			// from there. These ids must be kept constant if the data-bearing nodes
+			// change indexes, if for instance, they are scaled up and down.
+			//
+			replicaSetIndex = arbitersStartingIndex + processIndex
+		}
+
+		fcv := versions.CalculateFeatureCompatibilityVersion(b.mongodbVersion)
+		if b.fcv != "" {
+			fcv = b.fcv
+		}
 
+		// TODO: make a builder function for this struct
```

Comment:
Do we still want to add a builder as a part of this pr?

Reply:
Not as part of this PR. If you see in this function we have 2 ways of building 2 structs:

1. `Process` is built manually and then with some getters (SetPort, SetStoragePath)
2. `ReplicaSetMember` is built with a `newReplicaSetMember` functin, with parameters
3. The *standard* we have been doing is to use a Builder (like Secret.Builder)

The TODO is about the combination of strategies in this function; will add more context to the TODO in fact 👍